content/en/blog/_posts/2023-11-1-Gateway-API-Future.md

```diff
@@ -0,0 +1,257 @@
+When these are set, any generated infrastructure will have the provided labels and annotations set on them.
+
+For example, I may want to group all my resources for one application together:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: hello-world
+spec:
+  infrastructure:
+    labels:
+      app.kubernetes.io/name: hello-world
+```
+
+In the future, we are looking into more common infrastructure configurations, such as resource sizing.
+
+## Support for Websockets, HTTP/2 and more!
+
+Not all implementations of Gateway API support automatic protocol selection.
+In some cases protocols are disabled without an explicit opt-in. 
+
+When a Route's backend references a Kubernetes Service, application developers can specify the protocol using `ServicePort` [`appProtocol`][appProtocol] field.
+
+For example the following `store` Kubernetes Service is indicating the port `8080` supports HTTP/2 Prior Knowledge.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: store
+spec:
+  selector:
+    app: store
+  ports:
+  - protocol: TCP
+    appProtocol: kubernetes.io/h2c
+    port: 8080
+    targetPort: 8080
+```
+
+Currently, Gateway API has conformance testing for:
+
+- `kubernetes.io/h2c` - HTTP/2 Prior Knowledge
+- `kubernetes.io/ws` - WebSocket over HTTP
+
+[appProtocol]: https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol
+
+## `gwctl`, our new Gateway API command line tool
+
+`gwctl` is a command line tool that aims to be a `kubectl` replacement for viewing Gateway API resources.
+
+The initial release of `gwctl` that comes bundled with Gateway v1.0 release includes helpful features for managing Gateway API Policies.
+Gateway API Policies serve as powerful extension mechanisms for modifying the behaviour of Gateway resources.
+One challenge with using policies is that it may be hard to discover which policies are affecting which Gateway resources.
+`gwctl` helps bridge this gap by answering questions like:
+
+* Which policies are available for use in the Kubernetes cluster?
+* Which policies are attached to a particular Gateway, HTTPRoute, etc?
+* If policies are applied to multiple resources in the Gateway resource hierarchy, what is the effective policy that is affecting a particular resource? (For example, if an HTTP request timeout policy is applied to both an HTTPRoute and its parent Gateway, what is the effective timeout for the HTTPRoute?)
+
+`gwctl` is still in the very early phases of development and hence may be a bit rough around the edges.
+
+### Try it out!
+
+```bash
+# Clone the Gateway API repository.
+git clone https://github.com/kubernetes-sigs/gateway-api.git
+# Go to the gwctl directory.
+cd gateway-api/gwctl
+# Ensure vendor depedencies.
+go mod tidy
+go mod vendor
+# Build the gwctl binary.
+go build -o bin/gwctl cmd/main.go
+# Add binary to PATH.
+export PATH=./bin:${PATH}
+# Start using!
+gwctl help
+```
```

Comment:
I think this may be too much information for a blog post, I'd recommend linking to gwctl Readme at the end of the gwctl section for more information and removing this part.